### PR TITLE
Issue #3989: UTs should not use ROOT locale when they test violation/…

### PR DESCRIPTION
**Pull request by Luolc**
#3989 

Some information that might help:

`getMessageBundle ` in `BaseCheckTestSupport` refers to
https://github.com/checkstyle/checkstyle/blob/master/src/main/java/com/puppycrawl/tools/checkstyle/api/AbstractViolationReporter.java#L110
The resource bundle should be loaded with a correct path, which is placed in the same package as the check class. In most cases, UT is in the same package as the related check class, so we can just get the package by the test class itself. In other cases, they are in different packages, so we need to pass an extra class parameter as well.

This fix is to remove the ROOT locale in tests, and that's done. Meanwhile I have found quite a few hard-coded string literals which will lead to localization problem. Other changes have been made to fix them.

Diff report:
http://www.luolc.com/checkstyle-diff-report/issue3989/

### CODE REVIEW

File: src/main/resources/com/puppycrawl/tools/checkstyle/checks/indentation/messages_zh.properties [hunk 1]
```diff
@@ -1,5 +1,5 @@
 indentation.error.multi=''{0}'' 缩进了{1}个缩进符，应为以下缩进之一：{2}。
-indentation.child.error.multi=''{0}'' 子元素进了{1}个缩进符，应为以下缩进之一：{2}。
+indentation.child.error.multi=''{0}'' 子元素缩进了{1}个缩进符，应为以下缩进之一：{2}。
```
Comment (rnveach) [inline src/main/resources/com/puppycrawl/tools/checkstyle/checks/indentation/messages_zh.properties]: why is there change in property?

Comment (romani) [inline src/main/resources/com/puppycrawl/tools/checkstyle/checks/indentation/messages_zh.properties]: @Luolc , if that is fix for typo - please move it to separate PR.

File: src/test/java/com/puppycrawl/tools/checkstyle/checks/indentation/IndentationCheckTest.java [hunk 2]
```diff
@@ -160,6 +163,9 @@ private void verify(Configuration config, String filePath, String[] expected,
             final IndentComment... linesWithWarn) throws Exception {
         final Checker checker = createChecker(config);
         checker.addListener(new IndentAudit(linesWithWarn));
+        checker.setLocaleCountry(Locale.ROOT.getCountry());
+        checker.setLocaleLanguage(Locale.ROOT.getLanguage());
+        checker.finishLocalSetup();
```
Comment (romani) [inline src/test/java/com/puppycrawl/tools/checkstyle/checks/indentation/IndentationCheckTest.java]: We try to remove ROOT , why it was added there ?

Comment (Luolc) [inline src/test/java/com/puppycrawl/tools/checkstyle/checks/indentation/IndentationCheckTest.java]: The `IndentationCheckTest` is quite different from other tests. There are many `verifyWarns` in test methods. The actual assertion is at https://github.com/checkstyle/checkstyle/blob/master/src/test/java/com/puppycrawl/tools/checkstyle/checks/indentation/IndentationCheckTest.java#L1785-L1786, that to use an `endsWith` but not `equals` to judge the correctness.

The `getExpectedMessage()` is defined here, https://github.com/checkstyle/checkstyle/blob/master/src/test/java/com/puppycrawl/tools/checkstyle/checks/indentation/IndentationCheckTest.java#L1818-L1827
And the error message in English could be one of the followings:
```
indentation.error.multi=''{0}'' have incorrect indentation level {1}, expected level should be one of the following: {2}.
indentation.child.error.multi=''{0}'' child have incorrect indentation level {1}, expected level should be one of the following: {2}.
indentation.error=''{0}'' have incorrect indentation level {1}, expected level should be {2}.
indentation.child.error=''{0}'' child have incorrect indentation level {1}, expected level should be {2}.
```
The `endWith` way could work when the message is in English or Chinese. But in many other languages, it will not work at all. The messages could have different postfixes due to the difference of singular-plural or even pronounce stress.
i.e. in Finnish
```
indentation.error={0} sisennyssyvyydellä {1} ei ole oikealla syvyydellä {2}.
indentation.error.multi=''{0}'' on virheellinen sisennystason {1}, odottaa tasolla pitäisi olla yksi seuraavista: {2}.
indentation.child.error.multi=''{0}'' lapsi on väärä sisennystason {1}, odottaa tasolla pitäisi olla yksi seuraavista: {2}.
indentation.child.error=''{0}'' lapsi on väärä sisennystason {1}, odottaa taso olisi {2}.
```
We can see the messages have totally different postfixes.

I have logged what information the `IndentComment` really has and found that it doesn't have the token type info at all.  It only has the actual and expected intent.
So we can only know that the message is smth like `''{an unknown token type}'' have incorrect indentation level {actual intent}, expected level should be {expected intent}.` I guess it is why the original contributor didn't use `equals` but `endWith`.

To sum up, I haven't found another way to do the assertion without `endWith`. And if we are using `endWith`, it is not able to do the assertion in many non-English languages. Therefore, I use the old/ugly way to enforce the locale in English.

File: src/main/resources/com/puppycrawl/tools/checkstyle/checks/indentation/messages_zh.properties (hunk 1 above)
Comment (Luolc) [inline src/main/resources/com/puppycrawl/tools/checkstyle/checks/indentation/messages_zh.properties]: It is a typo. I tried to use `endWith` way (I describe it at https://github.com/checkstyle/checkstyle/pull/3993#discussion_r105829337) in Chinese at first and found that typo. The `endWith` way will not work if the message in Chinese has a wrong postfix.

File: src/test/java/com/puppycrawl/tools/checkstyle/checks/indentation/IndentationCheckTest.java (hunk 2 above)
Comment (rnveach) [inline src/test/java/com/puppycrawl/tools/checkstyle/checks/indentation/IndentationCheckTest.java]: @romani To do a 3-way validation for Indentation (expected vs message vs input file), the only way we could get the required indentation levels was to parse the numbers from the message directly. There is no hook in the check that lets us see these values before the message is created.
Here is the issue: https://github.com/checkstyle/checkstyle/issues/3105

The only way we can fix this and get the numbers outside the message is probably to either override `IndentationCheck` in test or to try and use powermock to call a test method before the check's method. We would have to target `IndentationCheck.indentationLog`.

Comment (romani) [inline src/test/java/com/puppycrawl/tools/checkstyle/checks/indentation/IndentationCheckTest.java]: I will split this to separate issue.

@rnveach , please approve PR . I am ready to merge.

Comment (romani) [inline src/test/java/com/puppycrawl/tools/checkstyle/checks/indentation/IndentationCheckTest.java]: new issue -  #4003
